File: dotnet/console/Console/Scenarios/AddNewDocumentVersionScenario.cs
using Lacuna.Signer.Api;
using Lacuna.Signer.Api.Documents;
using System;
using System.Collections.Generic;
using System.IO;
using System.Linq;
using System.Text;
using System.Threading.Tasks;

namespace Console.Scenarios
{
    class AddNewDocumentVersionScenario : Scenario
    {
        /**
         * This scenario demonstrates how to add a new document version to a existing document (the flow will be reset).
         */
        public override async Task RunAsync()
        {
            //1 - You need a document id
            var document = await CreateDocumentAsync();
            var docId = document.DocumentId;

            // 2. The file's bytes must be read by the application and uploaded
            var filePath = "sample.pdf";
            var fileName = Path.GetFileName(filePath);
            var file = File.ReadAllBytes(filePath);
            var uploadModel = await SignerClient.UploadFileAsync(fileName, file, "application/pdf");

            // 3. Define the name of the document which will be visible in the application
            var fileUploadModel = new FileUploadModel(uploadModel) { DisplayName = "Add New Document Version Sample" };

            // 4. Send the new version request
            var documentAddVersionRequest = new DocumentAddVersionRequest() { File = fileUploadModel };
            await SignerClient.AddDocumentVersionAsync(docId , documentAddVersionRequest);

        }
    }
}

File: nodejs/scenarios/addNewDocumentVersionScenario.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
const signer_node_client_1 = require("signer-node-client");
const scenario_1 = require("./scenario");
const signer_node_client_2 = require("signer-node-client");
const uploadApi = new signer_node_client_1.UploadApi(scenario_1.config);
const documentsApi = new signer_node_client_1.DocumentsApi(scenario_1.config);
(0, scenario_1.CreateDocument)().then((res) => {
    // 1. Retrieve document id
    const docId = res.documentId;
    // 2. The file's bytes must be read by the application and uploaded
    const filepath = "..\\samples\\sample.pdf";
    const fileName = "sample.pdf";
    const base64File = (0, signer_node_client_2.getBase64)(filepath);
    uploadApi
        .apiUploadsBytesPost({
        bytes: base64File,
    })
        .then((response) => {
        // 3. Define the name of the document which will be visible in the application
        const fileUploadModel = {
            name: fileName,
            displayName: "Add New Document Version Sample",
            contentType: "application/pdf",
            id: response.data.id,
        };
        // 4. Create object and send the new version request
        const documentAddVersionRequest = {
            file: fileUploadModel,
        };
        documentsApi
            .apiDocumentsIdVersionsPost(docId, documentAddVersionRequest)
            .then((response) => {
            console.log(response.status);
        });
    });
});
